app/src/main/java/net/nourepide/learning/cryptocurrencymonitor/MainListAdapter.kt

```diff
@@ -1,32 +1,34 @@
 package net.nourepide.learning.cryptocurrencymonitor
 
+import android.databinding.DataBindingUtil
+import android.databinding.ViewDataBinding
 import android.support.v7.widget.RecyclerView
 import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
+import net.nourepide.learning.cryptocurrencymonitor.MainListAdapter.MainViewHolder
+import net.nourepide.learning.cryptocurrencymonitor.entity.Cryptocurrency
 
-class MainListAdapter(private val viewModel: MainViewModel) : RecyclerView.Adapter<MainListAdapter.MainViewHolder>() {
-    override fun onCreateViewHolder(viewGroup: ViewGroup, value: Int) = MainViewHolder(
-        LayoutInflater.from(viewGroup.context)
-            .inflate(R.layout.item_cryptocurrency, viewGroup, false)
-    )
+class MainListAdapter(private val viewModel: MainViewModel) : RecyclerView.Adapter<MainViewHolder>() {
+    class MainViewHolder(private val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
```

Comment:
Вложенные классы должны идти в конце родительского класса, перед компаньоном.

[Котлиновцы рекомендуют](https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout) в некоторых случаях помещать его даже после компаньона, но мне кажется можно просто всегда ставить классы в конец родительского класса перед компаньоном и не париться.

*******

Цитата:

Generally, the contents of a class is sorted in the following order:

- Property declarations and initializer blocks
- Secondary constructors
- Method declarations
- Companion object

Do not sort the method declarations alphabetically or by visibility, and do not separate regular methods from extension methods. Instead, put related stuff together, so that someone reading the class from top to bottom would be able to follow the logic of what's happening. Choose an order (either higher-level stuff first, or vice versa) and stick to it.

Put nested classes next to the code that uses those classes. If the classes are intended to be used externally and aren't referenced inside the class, put them in the end, after the companion object.